### PR TITLE
network: clarify relationship between RA flags and DHCPv6 modes

### PR DESCRIPTION
**Pull request by alexhenrie**
In the documentation, using the term "managed" for both the RA flag and
the DHCPv6 mode is confusing because the mode is referred to as
"solicit" both in the official DHCPv6 documentation (see RFC 8415) and
in the WithoutRA option.

Furthermore, calling the other RA flag "other information" or "other
address configuration" is confusing because its official name is simply
"other configuration" (see RFC 4861 and RFC 5175) and it isn't used to
assign IP addresses.

Rewrite the documentation for DHCPv6Client and WithoutRA to make it
clear that getting the "managed" RA flag triggers the same kind of DHCP
request as WithoutRA=solicit, whereas getting the "other configuration"
RA flag triggers the same kind of DHCP request as
WithoutRA=information-request.

### CODE REVIEW

**Review by yuwata — APPROVED** [review]
LGTM. Thank you.

File: man/systemd.network.xml [hunk 1]
```diff
@@ -2172,9 +2172,9 @@ Table=1234</programlisting></para>
       <varlistentry>
         <term><varname>WithoutRA=</varname></term>
         <listitem>
-          <para>Allows DHCPv6 client to start without router advertisements's managed or other
-          address configuration flag. Takes one of <literal>no</literal>, <literal>solicit</literal>
-          or <literal>information-request</literal>. If this is not specified,
+          <para>Allows DHCPv6 client to start without router advertisements's "managed" or "other
```
Comment (yuwata) [inline man/systemd.network.xml]: Please use `<literal>` tag instead of `"`.

File: man/systemd.network.xml [hunk 2]
```diff
@@ -2172,9 +2172,9 @@ Table=1234</programlisting></para>
...
+          configuration" flag. Takes one of <literal>no</literal>, <literal>solicit</literal> or
```
Comment (yuwata) [inline man/systemd.network.xml]: This is not your fault, but could you add a comma after `<literal>solicit</literal>`?

File: man/systemd.network.xml (hunk 1 above)
Comment (alexhenrie) [reply man/systemd.network.xml]: Done.

File: man/systemd.network.xml (hunk 2 above)
Comment (alexhenrie) [reply man/systemd.network.xml]: Done.

**Comment by alexhenrie** [comment]
While we're at it, I just added another small commit to clarify the relationship between `DHCP=` and `WithoutRA=`. The previous documentation implied that all you have to do to get DHCPv6 working without RA is to set `DHCP=true` or `DHCP=ipv6`, but in reality `WithoutRA` must also be set to the desired DHCP mode.

**Comment by alexhenrie** [comment]
Sorry for the noise, I just noticed that I missed a quotation mark. They are all `<literal>` tags now.

And thanks for the speedy review!

**Comment by alexhenrie** [comment]
Thank you!

**Pull request merged**